Avoid infinite reminder comments

Makpoc opened the pull request:
The condition for diff > 24*Hours || diff < -24*Hours breaks when
crossing day bounderies. The date of the posted reminder is before the
day that it parses as deadline and thus it doesn't match when it's
searching for it at line 207.

When we ensure we are only looking for reminders for the same day we are
not allowing it to cross the day and will always match against the
current date

Fixes #14

Makpoc committed:
Issue-14 - Infinite reminder comments

The condition for diff > 24*Hours || diff < -24*Hours breaks when
crossing day bounderies. The date of the posted reminder is before the
day that it parses as deadline and thus it doesn't match when it's
searching for it at line 207.

When we ensure we are only looking for reminders for the same day we are
not allowing it to cross the day and will always match against the
current date

Signed-off-by: Makpoc <makpoc@gmail.com>

campoy commented on reminder/reminder.go:
```diff
@@ -196,7 +196,9 @@ func (c *InstallationClient) checkReminders(ctx context.Context, issue *issue) e
-			if diff := time.Until(reminder); diff > 24*time.Hour || diff < -24*time.Hour {
+			now := time.Now()
+			// not the same day
+			if reminder.Day() != now.Day() || reminder.Month() != now.Month() || reminder.Year() != now.Year() {
```
I'm not sure about this change, we not only care about the date, but also the time zone, right?

Makpoc replied:
Makes sense. Few lines above at L192 the reminder date is constructed in UTC TZ
```go
date = time.Date(date.Year(), date.Month(), date.Day(), 0, 0, 0, 0, time.UTC)
```
Looks to me like we can change the timezone of `now` on L199 to 
```go
now := time.Now().In(time.UTC)
```
Does that make sense?

Makpoc committed:
Infinite reminder comments

As per review comment - set UTC timezone as current tz before comparing
the dates.

Signed-off-by: Makpoc <makpoc@gmail.com>

campoy commented:
Thanks, @Makpoc !
